snuba/querylog/query_metadata.py

```diff
@@ -113,3 +114,13 @@ def status(self) -> QueryStatus:
         # If we do not have any recorded query and we did not specifically log
         # invalid_query, we assume there was an error somewhere.
         return self.query_list[-1].status if self.query_list else QueryStatus.ERROR
+
+
+class FailedRequestMetadata(ABC, SnubaQueryMetadata):
+    pass
+
+
+class RateLimitedRequestMetadata(FailedRequestMetadata):
+    @property
+    def status(self) -> QueryStatus:
+        return QueryStatus.RATE_LIMITED
```

Comment:
This is basically needed because, adding the quota control after parsing and before any query is processed introduces an issue with QueryMetadata. 
So far we had only two cases. Failures before parsing, which were dealt with individually. And failure after queries are produced. Everything else in between is considered an HTTP500.

Now we want to assign a status other than ERROR (500) to a query after parsing and before generating/executing any query.

So `RateLimitedRequestMetadata` captures this so we do not rely on the return status of a DB query (there is none) to record the status.